gapic/src/main/com/google/gapid/views/DeviceDialog.java

```diff
@@ -173,54 +173,89 @@ protected Control createDialogArea(Composite parent) {
+      Widgets.createTableColumn(compatibleDeviceTable,
+          "GPU", dev -> ((Device.Instance)dev).getConfiguration().getHardware().getGPU().getName());
+      Widgets.createTableColumn(compatibleDeviceTable,
+          "Driver version", dev -> Integer.toUnsignedString(((Device.Instance)dev).getConfiguration().getDrivers().getVulkan().getPhysicalDevices(0).getDriverVersion()));
+      Widgets.packColumns(compatibleDeviceTable.getTable());
+
+      compatibleDeviceTable.getTable().addListener(SWT.Selection, e -> {
+        runValidationCheck(getSelectedDevice());
       });
 
-      validationStatusLoader = widgets.loading.createWidgetWithImage(mainGroup,
+      // Validation widgets
+      validationStatusLoader = widgets.loading.createWidgetWithImage(compatibleGroup,
           widgets.theme.check(), widgets.theme.error());
       validationStatusLoader
-          .setLayoutData(withIndents(new GridData(SWT.LEFT, SWT.BOTTOM, false, false), 0, 0));
-      validationStatusText = createLink(mainGroup, "", e -> {
+          .setLayoutData(new GridData(SWT.LEFT, SWT.BOTTOM, false, false));
+      validationStatusText = createLink(compatibleGroup, "", e -> {
         Program.launch(URLs.DEVICE_COMPATIBILITY_URL);
       });
       validationStatusText.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false));
-
-      deviceCombo.getCombo().addListener(SWT.Selection, e -> {
-        runValidationCheck(getSelectedDevice());
+      validationStatusLoader.setVisible(false);
+      validationStatusText.setVisible(false);
+
+      // Use a group to keep the same look and feel for incompatible devices
+      Group incompatibleGroup = withLayoutData(
+          createGroup(composite, "Incompatible devices:", new GridLayout(1, false)),
+          new GridData(GridData.FILL_HORIZONTAL));
+      incompatibleDeviceTable = createTableViewer(incompatibleGroup, SWT.BORDER | SWT.SINGLE);
+      incompatibleDeviceTable.getTable().setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+
+      incompatibleDeviceTable.setContentProvider(ArrayContentProvider.getInstance());
+      incompatibleDeviceTable.setLabelProvider(new LabelProvider() {
+        @Override
+        public String getText(Object element) {
+          return Devices.getLabel(((IncompatibleDeviceInfo) element).device);
+        }
+      });
+      Widgets.createTableColumn(incompatibleDeviceTable, "Name", dev -> ((IncompatibleDeviceInfo)dev).device.getName());
+      Widgets.createTableColumn(incompatibleDeviceTable, "Serial", dev -> ((IncompatibleDeviceInfo)dev).device.getSerial());
+      Widgets.createTableColumn(incompatibleDeviceTable, "GPU", dev -> ((IncompatibleDeviceInfo)dev).device.getConfiguration().getHardware().getGPU().getName());
+      Widgets.createTableColumn(incompatibleDeviceTable, "Driver version", dev -> Integer.toUnsignedString(((IncompatibleDeviceInfo)dev).device.getConfiguration().getDrivers().getVulkan().getPhysicalDevices(0).getDriverVersion()));
```

Comment:
I think we should do a size check before `getPhysicalDevices(0)`, met out of boundary exception here sometimes.

Reply:
Done, in the new function Devices.GetDriverVersion()